oxen-io/session-android
# SS-2168 - User profile warnings

### Contributor checklist

 * Virtual Pixel 3a, Android 9 API 28
~~- [ ] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)~~

----------

### Description
This PR is to add user feedback in the following situations:
- Profile picture upload failed (Toast: "Failed to update profile"), and
- Profile picture removal failed (Toast: "Failed to remove display picture")

The OkHttp, conscrypt, and androidx core libraries were updated in the process of experimentation to add this functionality. Although the final result can be achieved without these updates, I see little reason to roll them back to our old, stale library versions when these fresh versions work just fine.

To test this PR you might like to:
- Put the device in Airplane mode and attempt to upload a profile picture - when it fails you'll see the toast "Failed to update profile", and
- With an existing profile picture, put the device in Airplane mode and then attempt to remove the profile picture. When the network sync of the profile picture removal fails the error toast "Failed to remove profile picture" will display.

Note: while we _could_ remove the local profile picture even if the network sync failed, this would be confusing and would also deny the user the chance to retry removal. Also, it could look **to them** as if they had no profile picture, but online to others they could - so it's best not to break that sync imho.
